### DISCUSSION TITLE
Expose waitForDedupingInterval() as part of the public API

### DISCUSSION THREAD

**Discussion by petehunt** [vercel/swr#2326] [Ideas]
Writing tests with SWR is very painful. If you don't know the magic incantation in #781, you can easily have nondeterministic tests that can be quite difficult to debug.

I'd expect there to be a way for a test to wait for all pending requests to complete. We have rolled our own at Twitter but I'd much prefer it to be documented, stable behavior rather than `sleep(1)` :)

**Comment by koba04** [comment]
`waitForDedupingInterval` that I've added depends on `sleep(1)` and  `testing-library`'s `act`, so it doesn't make sense to expose this as a public API.
But I think it's a good idea that SWR provides an API for the case.
SWR can provide a more stable API if it's implemented as a part of SWR because it can wait for a deduping interval and async mutations precisely.

The downside is to increase the file size even though it's an API for testing.
But I think it worth considering adding it even though I have no idea what the appropriate interface is.

**Comment by mAAdhaTTah** [comment]
@koba04 If you're concerned about file size, you could provide a `swr/testing` module to export them through so they don't get included as part of the main bundle.
